File: packages/core/src/actions/ActionResult.ts
export interface ActionResult {
  actionId: string;
  success: boolean;
  result?: unknown;
  error?: string;
  timestamp: string;
}

export function createActionResult(
  actionId: string,
  success: boolean,
  result?: unknown,
  error?: string
): ActionResult {
  return {
    actionId,
    success,
    result,
    error,
    timestamp: new Date().toISOString(),
  };
}

File: packages/swift/Sources/MobileAgentOS/Actions/ActionResult.swift
import Foundation

/// Result of action execution
public struct ActionResult: Codable {
    public let actionId: String
    public let success: Bool
    public let message: String
    public let timestamp: Date
    
    public init(
        actionId: String,
        success: Bool,
        message: String
    ) {
        self.actionId = actionId
        self.success = success
        self.message = message
        self.timestamp = Date()
    }
}